sympy/solvers/solveset.py

```diff
@@ -978,6 +985,318 @@ def _solveset(f, symbol, domain, _check=False):
+        -3*log(2)
+    {------------------}
+     -2*log(3) + log(2)
+    >>> pprint(tsolve(f, x), use_unicode=False)
+         /   3     \
+         | --------|
+         | log(2/9)|
+    [-log\2         /]
+
+    2) Extensible
+
+    The API of `\_transolve` is designed such that it is easily
+    extensible, i.e. the code that solves a given class of
+    equations is encapsulated in a helper and not mixed in with
+    the code of \_transolve itself.
+
+    3) Modular
+
+    `\_transolve` is designed to be modular i.e, for every class of
+    equation a separate helper for identification and solving is
+    implemented. This makes it easy to change or modify any of the
+    method implemented directly in the helpers without interfering
+    with the actual structure of the API.
+
+    4) Faster Computation
+
+    Solving equation via `\_transolve` is much faster as compared to
+    `\_tsolve`. In `solve` attempts are made computing every possibility
+    to get the solutions. This series of attempts makes solving a bit
+    slow. Whereas in `\_transolve` computation begins only when the
+    equation is identified of being a particular type.
+
+    How to add new class of equations
+    =================================
+
+    `\_transolve` is designed in such a way that it becomes an easy
+    task to add a new class of equation solver.
+
+    The first task that needs to be done for adding your own solver
+    is to decide from where its *identification* *helper* will be
+    invoked. To do so determine the general form of the class of the
+    equation, for example the general form of the exponential equation
+    is `a*f(x) + b*g(x)` where `f(x)` and `g(x)` are power terms, so we
+    need to place the invocation condition inside `Add` case. Once the
+    place for *identification* *helper* is determined, you can add a
+    call to its *solving* *helper*.
+
+    For your class of equation you need to define your own
+    *identification* and *solving* *helpers*. The *identification*
+    *helper* should be implemented for generalised cases and should
+    return either `True` if the given equation belongs to the class
+    otherwise `False`. *Solving* *helpers* needs to be implemented with
+    heuristics or algorithms that solve a majority of the equations
+    belonging to that class. The value returned could be either the
+    exact solution or a reduced form of the equation which `solveset`
+    can handle.
+
+    Apart from this, a few other things needs to be taken care while
+    adding an equation solver:
+
+    - Naming conventions:
+      Name of the *identification* *helper* should be as
+      `\_check\_class` where `class` will be the name or abbreviation
+      of the class of equation. The *solving* *helper* will be named as
+      `\_solve\_class`.
+      For example: for exponential equations it becomes `\_check\_expo`
+      and `\_solve\_expo`.
+    - *Identification* *helpers* should take two input parameters,
```

Comment:
Is this point okay for the API design of the helpers?
Ref: [this](https://github.com/sympy/sympy/pull/14736#discussion_r198129763) comment